sql/schemachanger: support ADD COLUMN SERIAL for DSC unique_rowid only

Dedej-Bergin opened the pull request:
Previously we could only add SERIAL columns with the legacy schema changer
with these code changes the ALTER TABLE ADD COLUMN statement
can now add SERIAL type columns via the declarative schema changer for
the default rowid serial_normalization mode.

Informs: https://github.com/cockroachdb/cockroach/issues/126900
Release note: none

Dedej-Bergin committed:
sql/schemachanger: support ADD COLUMN SERIAL for DSC unique_rowid only

Previously we could only add SERIAL columns with the legacy schema changer
with these code changes the ALTER TABLE ADD COLUMN statement
can now add SERIAL type columns via the declarative schema changer for
the default rowid serial_normalization mode.

Informs: #126900
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/129377)
<!-- Reviewable:end -->

fqazi reviewed (changes requested):
Just some very minor comments. Great work!

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-:-O5u3Ckb0HzodkJ2xJa7:b9w23j)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Dedej-Bergin)

___
*[`pkg/sql/serial.go` line 170 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u25FX2g3IJZYjf1qK:-O5u25FX2g3IJZYjf1qL:bkeaqj8) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/serial.go#L170)):*
> ```Go
> 						"upgrading the column %s to %s to utilize the session serial_normalization setting",
> 						d.Name.String(),
> 						types.Int.SQLString(),
> ```

Not sure why this change was made?
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 83 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u2JELB8lWmXn5D2KD:-O5u2JELB8lWmXn5D2KE:b6d5x4l) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L83)):*
> ```Go
> 		}
> 		d = alterTableAddColumnSerial(b, d, tn)
> 		d.IsSerial = false
> ```

Why do we even need to flip this?
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 297 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u2noo5KK7IYJBhLJe:-O5u2noo5KK7IYJBhLJf:b-gvo7po) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L297)):*
> ```Go
> // UseFullWidthIntType For unique_rowid(), unordered_unique_rowid(), or virtual sequences,
> // full-width integer type is used when the values will not fit otherwise.
> func useFullWidthIntType(ctx BuildCtx, d *tree.ColumnTableDef, sender eval.ClientNoticeSender) {
> ```

Nit: Does this need to be a function? There is no-reuse here or clarity improvement.
___
*[`pkg/sql/logictest/testdata/logic_test/alter_table` line 3995 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u3BHL6ZLFo9yu0xqa:-O5u3BHL6ZLFo9yu0xqb:b-tkkicz) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/logictest/testdata/logic_test/alter_table#L3995)):*
> ```
> 
> statement ok
> SET serial_normalization = sql_sequence
> ```

Should we check other modes too?


<!-- Sent from Reviewable.io -->

Dedej-Bergin reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129377)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi)

___
*[`pkg/sql/serial.go` line 170 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u25FX2g3IJZYjf1qK:-O5xWHHtC_uMuN3z0wwk:bff2ps8) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/serial.go#L170)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Not sure why this change was made?
</blockquote></details>

It's leftovers from a failed refactoring attempt, thanks for catching it! Just updated it
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 83 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u2JELB8lWmXn5D2KD:-O5xQRPK97-gedsksSfG:b-8p0rry) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L83)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Why do we even need to flip this?
</blockquote></details>

The d.IsSerial was already being flipped before, it was just happening a few function/method calls deep which made it a bit unclear.

But we need d.IsSerial to be false because it gets checked again lower in the code:

On line 115: cdd, err := tabledesc.MakeColumnDefDescs(b, d, b.SemaCtx(), b.EvalCtx(), tree.ColumnDefaultExprInAddColumn)

This MakeColumnDefDescs has this logic:



_Code snippet:_
```Go
func MakeColumnDefDescs(
	ctx context.Context,
	d *tree.ColumnTableDef,
	semaCtx *tree.SemaContext,
	evalCtx *eval.Context,
	defaultExprCtx tree.SchemaExprContext,
) (*ColumnDefDescs, error) {
	if d.IsSerial {
		// To the reader of this code: if control arrives here, this means
		// the caller has not suitably called processSerialInColumnDef()
		// prior to calling MakeColumnDefDescs. The dependent sequences
		// must be created, and the SERIAL type eliminated, prior to this
		// point.
		return nil, pgerror.New(pgcode.FeatureNotSupported,
			"SERIAL cannot be used in this context")
	}
```
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 297 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u2noo5KK7IYJBhLJe:-O5xU9hVCuQZPF9FhLAW:bgs79qr) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L297)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Nit: Does this need to be a function? There is no-reuse here or clarity improvement.
</blockquote></details>

Thanks for catching this, it does not need to be a function, I just made the changes for this.
___
*[`pkg/sql/logictest/testdata/logic_test/alter_table` line 3995 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u3BHL6ZLFo9yu0xqa:-O5xTANz6Tx-9bpRPX0R:b-fpj4jx) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/logictest/testdata/logic_test/alter_table#L3995)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Should we check other modes too?
</blockquote></details>

Yes, but maybe in future PRs?  I'm planning on splitting this epic into 3 PRs, and I'll be re-writing/updating this testcase with the next PR as I will add support for sql_sequence.  This is a temporary testcase, when we close this epic we won't need this specific testcase as sql_sequence and other modes will work and will not return an error.


<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
It looks good to me too. I had a refactor suggestion and some other tests that we may want.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-:-O5xc5DW5KVM7RXXL-Yn:bukpjzx)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Dedej-Bergin and @fqazi)

___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 80 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5xaaXxBcIVO8twnwIf:-O5xaaXxBcIVO8twnwIg:b-fdj3ap) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L80)):*
> ```Go
> 	if d.IsSerial {
> 		if b.SessionData().SerialNormalizationMode != sessiondatapb.SerialUsesRowID {
> 			panic(scerrors.NotImplementedErrorf(d, "contains serial data type in unsupported mode"))
> ```

nit: I suggest moving this panic into `alterTableAddColumnSerial`. Additionally, consider adding a switch statement to handle the different modes, which would be a logical place for this change.
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 286 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5x_XGX7a82okN4AmXk:-O5x_XGX7a82okN4AmXl:b-cihoi8) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L286)):*
> ```Go
> 		defType.Name(), b.SessionData().SerialNormalizationMode.String()))
> 
> 	if defType.Width() < types.Int.Width() {
> ```

nit: I believe this logic is specific to using row ID as the default expression. And seeing as you need to expand this function for other serial normalization modes, could we structure this in a switch statement? That will make it easier to expand for other modes.
___
*[`pkg/sql/logictest/testdata/logic_test/alter_table` line 3971 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5xbJ7O4HLqtMtVGiKi:-O5xbJ7O4HLqtMtVGiKj:b19cngw) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/logictest/testdata/logic_test/alter_table#L3971)):*
> ```
> subtest end
> 
> subtest alter_table_add_column_serial
> ```

Any tests to verify the dsc plan? i.e. in pkg/sql/schemachanger/scbuild/testdata, pkg/sql/schemachanger/scplan/testdata, and/or pkg/sql/schemachanger/testdata/end_to_end
___
*[`pkg/sql/logictest/testdata/logic_test/alter_table` line 3977 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5xb16-5XdugKiooleV:-O5xb16-5XdugKiooleW:b38j703) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/logictest/testdata/logic_test/alter_table#L3977)):*
> ```
> 
> statement ok
> create table roach (id int);
> ```

Can we add rows to the table? I think the alter is more interesting if it has to generate values for the new column.


<!-- Sent from Reviewable.io -->

Dedej-Bergin reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129377)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi and @spilchen)

___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 80 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5xaaXxBcIVO8twnwIf:-O5xpFmhBFUMFMs9el0R:bcoobku) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L80)):*
<details><summary><i>Previously, spilchen wrote…</i></summary><blockquote>

nit: I suggest moving this panic into `alterTableAddColumnSerial`. Additionally, consider adding a switch statement to handle the different modes, which would be a logical place for this change.
</blockquote></details>

Thanks Matt! I was thinking of moving the panic and adding the switch on my next PR as the code grows with additional mode support.
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 286 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5x_XGX7a82okN4AmXk:-O5xpdAI7akU2a9dyWX4:b-qm2wcd) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L286)):*
<details><summary><i>Previously, spilchen wrote…</i></summary><blockquote>

nit: I believe this logic is specific to using row ID as the default expression. And seeing as you need to expand this function for other serial normalization modes, could we structure this in a switch statement? That will make it easier to expand for other modes.
</blockquote></details>

Thanks Matt! I'll add the switch on my next PR if that's okay? As the code grows with additional mode support.
___
*[`pkg/sql/logictest/testdata/logic_test/alter_table` line 3971 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5xbJ7O4HLqtMtVGiKi:-O5xq-3Q38ByCECMklKb:bd2nlkw) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/logictest/testdata/logic_test/alter_table#L3971)):*
<details><summary><i>Previously, spilchen wrote…</i></summary><blockquote>

Any tests to verify the dsc plan? i.e. in pkg/sql/schemachanger/scbuild/testdata, pkg/sql/schemachanger/scplan/testdata, and/or pkg/sql/schemachanger/testdata/end_to_end
</blockquote></details>

Just added them! Thanks Matt!
___
*[`pkg/sql/logictest/testdata/logic_test/alter_table` line 3977 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5xb16-5XdugKiooleV:-O5xr9YV7mTHqOQJro-e:b-l6gtbk) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/logictest/testdata/logic_test/alter_table#L3977)):*
<details><summary><i>Previously, spilchen wrote…</i></summary><blockquote>

Can we add rows to the table? I think the alter is more interesting if it has to generate values for the new column.
</blockquote></details>

Sounds good, I added some rows, thanks Matt!


<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> Reviewed 2 of 2 files at r2, 1 of 1 files at r3, 15 of 15 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129377)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi)

___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 80 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5xaaXxBcIVO8twnwIf:-O61I36M4wAR-_Iqwtjz:blr00t6) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L80)):*
<details><summary><i>Previously, Dedej-Bergin (Bergin Dedej) wrote…</i></summary><blockquote>

Thanks Matt! I was thinking of moving the panic and adding the switch on my next PR as the code grows with additional mode support.
</blockquote></details>

Yes, this is totally fine by me.
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 83 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u2JELB8lWmXn5D2KD:-O61IN-aEeHbV39HNw-Z:b-sxomro) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L83)):*
<details><summary><i>Previously, Dedej-Bergin (Bergin Dedej) wrote…</i></summary><blockquote>

The d.IsSerial was already being flipped before, it was just happening a few function/method calls deep which made it a bit unclear.

But we need d.IsSerial to be false because it gets checked again lower in the code:

On line 115: cdd, err := tabledesc.MakeColumnDefDescs(b, d, b.SemaCtx(), b.EvalCtx(), tree.ColumnDefaultExprInAddColumn)

This MakeColumnDefDescs has this logic:
</blockquote></details>

I would suggest that we leave a comment so that we know why we flip it. Also, could we do the flip inside `alterTableAddColumnSerial`? I think that makes sense logically since we are gong to be returning a modified `ColumnTableDef` anyway.
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 286 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5x_XGX7a82okN4AmXk:-O61Int494L7kmeX3oBN:ba5ezuo) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L286)):*
<details><summary><i>Previously, Dedej-Bergin (Bergin Dedej) wrote…</i></summary><blockquote>

Thanks Matt! I'll add the switch on my next PR if that's okay? As the code grows with additional mode support.
</blockquote></details>

Yes, I'm okay with this. Thanks.


<!-- Sent from Reviewable.io -->

Dedej-Bergin commented:
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 83 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O5u2JELB8lWmXn5D2KD:-O61ez7s6KEYI_6gJ5fL:bfkgf00) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b1946db3ff72ef48b3bb0e55ddb2edf44b67320/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L83)):*
<details><summary><i>Previously, spilchen wrote…</i></summary><blockquote>

I would suggest that we leave a comment so that we know why we flip it. Also, could we do the flip inside `alterTableAddColumnSerial`? I think that makes sense logically since we are gong to be returning a modified `ColumnTableDef` anyway.
</blockquote></details>

Sounds good, I added the comment and moved it to alterTableAddColumnSerial.


<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129377)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Dedej-Bergin and @fqazi)

___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 301 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O62AZmjCOVOvUYMIT9n:-O62AZmjCOVOvUYMIT9o:b1q82va) ([raw file](https://github.com/cockroachdb/cockroach/blob/a25d42579a3e63bba0d6d22ce944a7bcf4d5c2de/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L301)):*
> ```Go
> 	}
> 
> 	// Flip to false, as we validate or panic with this in MakeColumnDefDescs.
> ```

minor nit: Would it be better to phrase this in terms of what we're doing rather than what we're avoiding? For example:
```
// Serial is an alias for a real column definition. Set to false to indicate the alias has been remapped.
```

I also noticed that in the function comment for `MakeColumnDefDescs`, we specify which functions should be called to clear the `IsSerial` flag. Should we also mention `alterTableAddColumnSerial` for DSC alters?


<!-- Sent from Reviewable.io -->

Dedej-Bergin commented:
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 301 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O62AZmjCOVOvUYMIT9n:-O62QuYS4TS_lihCxmXC:b-bu0mh4) ([raw file](https://github.com/cockroachdb/cockroach/blob/a25d42579a3e63bba0d6d22ce944a7bcf4d5c2de/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L301)):*
<details><summary><i>Previously, spilchen wrote…</i></summary><blockquote>

minor nit: Would it be better to phrase this in terms of what we're doing rather than what we're avoiding? For example:
```
// Serial is an alias for a real column definition. Set to false to indicate the alias has been remapped.
```

I also noticed that in the function comment for `MakeColumnDefDescs`, we specify which functions should be called to clear the `IsSerial` flag. Should we also mention `alterTableAddColumnSerial` for DSC alters?
</blockquote></details>

Yes, I just changed the comment. For the function comment for MakeColumnDefDescs I was thinking of trying to refactor it in a later PR.  So maybe for now I will leave it as is and I will either refactor it or update the comment for it in a future PR if that's okay?


<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 2 files at r2, 4 of 15 files at r4, 1 of 2 files at r5, 12 of 12 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-:-O62oocK9BW5Qr6lv6rz:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

Dedej-Bergin commented:
bors r+

spilchen reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-:-O65xAs33FtbaRUO6WjD:bnfp4nl)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @fqazi)

___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go` line 301 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/129377#-O62AZmjCOVOvUYMIT9n:-O65wtRI7yjsyGnLAu5p:b-yzreb6) ([raw file](https://github.com/cockroachdb/cockroach/blob/a25d42579a3e63bba0d6d22ce944a7bcf4d5c2de/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go#L301)):*
<details><summary><i>Previously, Dedej-Bergin (Bergin Dedej) wrote…</i></summary><blockquote>

Yes, I just changed the comment. For the function comment for MakeColumnDefDescs I was thinking of trying to refactor it in a later PR.  So maybe for now I will leave it as is and I will either refactor it or update the comment for it in a future PR if that's okay?
</blockquote></details>

Yes, that's fine. Thanks for revising.


<!-- Sent from Reviewable.io -->

celiala commented:
bors crashed yesterday. re-trying:

bors r+
